data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/AbstractBuffer.java

```diff
@@ -142,6 +148,16 @@ public Map.Entry<Collection<T>, CheckpointState> read(int timeoutInMillis) {
         return readResult;
     }
 
+    protected void updateLatency(Collection<T> records) {
+        for (T rec : records) {
+            if (rec instanceof Record) {
+                Record<Event> record = (Record<Event>)rec;
```

Comment:
Not all buffers hold records at the moment.

We should change these lines to support non-events. Also, I think you don't need to cast to `DefaultEventHandle`.

```
Object data = rec.getData();
if(data instanceof Event) {
  Event event = (Event) data;
  Instant receivedTime = event.getEventHandle().getInternalOriginationTime();
  latencyTimer.record(Duration.between(receivedTime, Instant.now()));
}
```